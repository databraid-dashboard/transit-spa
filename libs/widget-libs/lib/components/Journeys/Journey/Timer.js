var _jsxFileName = '../../src/components/Journeys/Journey/Timer.jsx';

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

import React, { Component } from 'react';
import PropTypes from 'prop-types';

var Timer = function (_Component) {
  _inherits(Timer, _Component);

  function Timer(props) {
    _classCallCheck(this, Timer);

    var _this = _possibleConstructorReturn(this, (Timer.__proto__ || Object.getPrototypeOf(Timer)).call(this, props));

    _this.state = { currentCount: props.seconds };
    return _this;
  }

  _createClass(Timer, [{
    key: 'componentDidMount',
    value: function componentDidMount() {
      var _this2 = this;

      this.timerID = setInterval(function () {
        return _this2.decrement();
      }, 1000);
    }
  }, {
    key: 'componentWillReceiveProps',
    value: function componentWillReceiveProps(nextProps) {
      this.setState({ currentCount: nextProps.seconds });
    }
  }, {
    key: 'componentWillUnmount',
    value: function componentWillUnmount() {
      clearInterval(this.timerID);
    }
  }, {
    key: 'decrement',
    value: function decrement() {
      var newCount = this.state.currentCount - 1;

      if (newCount >= 1) {
        this.setState({ currentCount: newCount });
      }

      if (newCount === 0) {
        this.props.onComplete();
        clearInterval(this.timerID);
      }
    }
  }, {
    key: 'render',
    value: function render() {
      var currentCount = this.state.currentCount;


      var sec = currentCount % 60;
      var min = (currentCount - sec) / 60;

      var seconds = sec < 10 ? '0' + sec : sec;
      var minutes = min < 10 ? '0' + min : min;

      return React.createElement(
        'div',
        {
          __source: {
            fileName: _jsxFileName,
            lineNumber: 45
          },
          __self: this
        },
        minutes,
        ':',
        seconds
      );
    }
  }]);

  return Timer;
}(Component);

Timer.propTypes = {
  seconds: PropTypes.number.isRequired,
  onComplete: PropTypes.func.isRequired
};

Timer.defaultProps = {
  seconds: 6,
  onComplete: function onComplete() {}
};

export default Timer;