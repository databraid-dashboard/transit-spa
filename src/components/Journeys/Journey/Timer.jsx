import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Timer extends Component {
  constructor(props) {
    super(props);
    this.state = { count: props.seconds, expired: false };
  }

  decrementOrExpire() {
    return this.state.count > 0 ? { count: this.state.count - 1 } : { expired: true };
  }

  render() {
    const { count } = this.state;

    setTimeout(() => this.setState(this.decrementOrExpire), 1000);

    const sec = count % 60;
    const seconds = sec < 10 ? `0${sec}` : sec;
    const min = (count - seconds) / 60;
    const minutes = min < 10 ? `0${min}` : min;

    return this.state.expired
      ? this.props.onComplete()
      : <div>
        {minutes}:{seconds}
      </div>;
  }
}

Timer.propTypes = {
  seconds: PropTypes.number.isRequired,
  onComplete: PropTypes.func.isRequired,
};

Timer.defaultProps = {
  seconds: 6,
  onComplete: () => {},
};

export default Timer;
