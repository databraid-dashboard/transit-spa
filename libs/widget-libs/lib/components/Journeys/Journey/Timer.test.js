var _jsxFileName = '../../src/components/Journeys/Journey/Timer.test.jsx',
    _this = this;

import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Timer, { decrement, onComplete } from './Timer';

describe('Timer', function () {
  var callback = function callback() {};

  it('renders minutes and seconds when passed seconds as props', function () {
    var component = shallow(React.createElement(Timer, { seconds: 120, onComplete: callback, __source: {
        fileName: _jsxFileName,
        lineNumber: 10
      },
      __self: _this
    }));
    expect(toJson(component)).toMatchSnapshot();
  });

  it('renders double-digit minutes and seconds when passed seconds as props', function () {
    var component = shallow(React.createElement(Timer, { seconds: 600, onComplete: callback, __source: {
        fileName: _jsxFileName,
        lineNumber: 15
      },
      __self: _this
    }));
    expect(toJson(component)).toMatchSnapshot();
  });

  it('has decrement method', function () {
    var component = shallow(React.createElement(Timer, { seconds: 120, onComplete: callback, __source: {
        fileName: _jsxFileName,
        lineNumber: 20
      },
      __self: _this
    }));
    expect(component.instance().decrement());
  });

  it('accepts a callback function as props', function () {
    var component = mount(React.createElement(Timer, { seconds: 120, onComplete: callback, __source: {
        fileName: _jsxFileName,
        lineNumber: 25
      },
      __self: _this
    }));
    expect(component.props().onComplete).toEqual(callback);
  });

  it('initializes with currentCount equal to the seconds passed in as props', function () {
    var component = mount(React.createElement(Timer, { seconds: 120, onComplete: callback, __source: {
        fileName: _jsxFileName,
        lineNumber: 30
      },
      __self: _this
    }));
    expect(component.state('currentCount')).toEqual(120);
  });

  it('calls decrement and decreases the currentCount', function () {
    var component = shallow(React.createElement(Timer, { seconds: 120, onComplete: callback, __source: {
        fileName: _jsxFileName,
        lineNumber: 35
      },
      __self: _this
    }));
    expect(component.instance().state.currentCount).toEqual(120);
    component.instance().decrement();
    expect(component.instance().state.currentCount).toEqual(119);
  });

  it('has componentWillUnmount method', function () {
    var component = shallow(React.createElement(Timer, { seconds: 120, onComplete: callback, __source: {
        fileName: _jsxFileName,
        lineNumber: 42
      },
      __self: _this
    }));
    expect(component.instance().componentWillUnmount());
  });

  it('has componentDidMount method', function () {
    var component = shallow(React.createElement(Timer, { seconds: 120, onComplete: callback, __source: {
        fileName: _jsxFileName,
        lineNumber: 47
      },
      __self: _this
    }));
    expect(component.instance().componentDidMount());
  });

  it('has componentWillReceiveProps method', function () {
    var component = shallow(React.createElement(Timer, { seconds: 120, onComplete: callback, __source: {
        fileName: _jsxFileName,
        lineNumber: 52
      },
      __self: _this
    }));
    expect(component.instance().state.currentCount).toEqual(120);
    component.instance().componentWillReceiveProps({ seconds: 300 });
    expect(component.instance().state.currentCount).toEqual(300);
  });

  it('calls onComplete when decrementing to 0', function () {
    var onComplete = jest.fn();
    var component = mount(React.createElement(Timer, { seconds: 1, loading: true, onComplete: onComplete, __source: {
        fileName: _jsxFileName,
        lineNumber: 60
      },
      __self: _this
    }));
    component.instance().decrement();
    expect(onComplete).toHaveBeenCalled();
  });
});