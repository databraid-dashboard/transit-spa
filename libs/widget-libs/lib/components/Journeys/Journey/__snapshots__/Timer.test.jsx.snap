// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timer renders double-digit minutes and seconds when passed seconds as props 1`] = `
<div>
  10
  :
  00
</div>
`;

exports[`Timer renders minutes and seconds when passed seconds as props 1`] = `
<div>
  02
  :
  00
</div>
`;
